restock.lic

```diff
@@ -37,10 +38,13 @@ class Restock
       item['buy_num'] = buy_num
       items_to_restock.push(item)
     end
-    ### The following line is for the bribe to the night attendant to let you in.
-    coin_needed += 2502 if UserVars.sun['night'] && town == /\bShard\b/
-    get_money_from_bank("#{coin_needed} copper", "#{town_currency(town)}", "#{town}") if coin_needed > 0
+    if coin_needed > 0
+      ### The following line is for the bribe to the night attendant to let you in.
+      coin_needed += 2502 if UserVars.sun['night'] && town =~ /\bShard\b/
+      get_money_from_bank("#{coin_needed} copper", "#{town_currency(town)}", "#{town}")
+    end
     items_to_restock.each do |item|
+      purchased = true
```

Comment:
This could be moved outside the loop. Just above:
```ruby
purchase = true unless items_to_restock.empty?
items_to_restock.each do |item|
  etc...
```

Reply:
Testing real quick.